src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs

```diff
@@ -82,6 +82,24 @@ internal class SocketConnectivitySubchannelTransport : ISubchannelTransport, IDi
     private object Lock => _subchannel.Lock;
     public BalancerAddress? CurrentAddress => _currentAddress;
     public TimeSpan? ConnectTimeout { get; }
+    public TransportStatus TransportStatus
+    {
+        get
+        {
+            lock (Lock)
+            {
+                if (_initialSocket != null)
+                {
+                    return TransportStatus.InitialSocket;
+                }
+                if (_activeStreams.Count > 0)
+                {
+                    return TransportStatus.ActiveStream;
+                }
+                return TransportStatus.Unknown;
```

Comment:
Is it actually unknown or does this mean uninitialized?

Reply:
I renamed it to `NotConnected`. There are a bunch of reasons to be in that state. `NotConnected` name fits them all.

Reply:
Much clearer.  Thanks!